dracut/30ignition/ignition-reboot.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+set -euo pipefail
+
+# Checks if kernel argument directory exists,
+# then redeploy and reboot the system if it exists
+reboot_if_kargs_dir_exists() {
+    if [ -d /etc/ostree/kargs.d ]
+    then
+        ostree admin deploy fedora/x86_64/coreos/testing-devel
```

Comment:
Shouldn't have code that is specific to a hardcoded ref.  Though this of course gets into a bit of a mess as all the nice high level logic around kargs is most recently in *rpm-ostree*, specifically `rpm-ostree kargs` which handles all of this.

Which isn't designed to be run from the initramfs today, although for that matter neither is ostree.

Are you sure this script would work?  It looks like you're using `/etc/ostree/kargs.d` but that's the copy in the *initramfs*, it'd need to be `/sysroot/etc/ostree/kargs.d` here right?

Reply:
> Shouldn't have code that is specific to a hardcoded ref.
> although for that matter neither is ostree.

Can I use `ostree refs --repo /sysroot/ostree/repo` to retrieve the `REFSPEC` or `ostree` should not be used at all?

> Are you sure this script would work? 

I'm not sure, since I haven't tested on this.. will be testing tomorrow morning

> It looks like you're using /etc/ostree/kargs.d but that's the copy in the initramfs, it'd need to be /sysroot/etc/ostree/kargs.d

I have a blurred understanding of which part is in initramfs and which part is in sysroot before. Now I think I have a sense that before mounting sysroot, all of the files resides in _initramfs_. Thank you for pointing that out !

Reply:
Note also you'll want to test this on top of https://github.com/ostreedev/ostree/pull/1836. See some tips from @cgwalters and @rfairley on doing that in https://github.com/ostreedev/ostree/issues/479.

> Can I use `ostree refs --repo /sysroot/ostree/repo` to retrieve the `REFSPEC` or `ostree` should not be used at all?

Sure. There shouldn't be other refs or deployments in there on first boot normally. Though another approach that would survive that would be to pick up the `ostree=` karg (see e.g. `cmdline_arg` in the generator), then pickup `refspec` from `$(realpath $ostree_arg).origin`. Not necessary as a first pass though!

Reply:
:+1: - I left some instructions with those tips collected here: https://github.com/ostreedev/ostree/pull/1836#issuecomment-495809370

To test with Ignition, instead of editing `fedora-coreos-base.yaml` and adding kargs snippets using the `add-files` field, you can add those files under `/etc/ostree/kargs.d` through an Ignition config (haven't tried this, but should work!).

Reply:
Using `ostree refs --repo /sysroot/ostree/repo` for now, might change to `$(realpath $ostree_arg).origin` in the future